### PR TITLE
Refactored AIP main modules using classes, defaults, and enums.

### PR DESCRIPTION
**Pull request by markich**
#### `Brief Description`:

- **WHAT**:

1. Updated modules and behaviour to start using well defined classes, recognizable defaults, and enums.
2. Implemented typing so as to define methods as contracts.
3. Removed biased language (whitelist -> safelist | blacklist -> blocklist | master -> main)
3. Implemented logging using  **__ builtins __**  package.
4. Added configuration files: **Pipfile**, **requirements.txt**, and **tox.ini**.
5. Implemented tests for **AIP**, **Flow**, **Methodology**, **MetricWeights**, **RawRating**, and **Safelist** using pytest.
6. Set **tox.ini** to provide support to implement CI/CD pipelines with **GitHub Actions** or **CircleCI**.
7. Added **setup.py** to provide support to release **AIP** as a **PyPI** dependency.

- **HOW**: Created the following classes **AIP**, **Flow**, **Methodology**, **MetricWeights**, **RawRating**, and **Safelist**.

- **WHY**: In order to improve maintainability, readability, and future re-classing pivoting.

- **UP NEXT**:

1. Implement test for AIP using mocked payloads from real data.

#### `Test Validation`:
In order to run involved tests, do:
```
python -m pytest tests/
python -m pytest tests/test_aip.py
python -m pytest tests/test_flows.py
python -m pytest tests/test_methodology.py
python -m pytest tests/test_metric_weights.py
python -m pytest tests/test_raw_ratings.py
python -m pytest tests/test_safelist.py
```
##### Test Suite
test_suite_2022-02-01

##### Test `AIP`
test_aip_2022-02-01

##### Test `Flow`
test_flow_2022-02-01

##### Test `Methodology`
test_methodology_2022-02-01

##### Test `MetricWeights`
test_metric_weights_2022-02-01

##### Test `RawRating`
test_raw_rating_2022-02-01

##### Test `Safelist`
test_safelist_2022-02-01

### CODE REVIEW

**Comment by eldraco** [comment]
This is great @markich ! Thanks a lot for the work and effort. We are finishing a paper now and then we will talk to you to merge this. We need to test that our systems keep working in the same way with your modifications. 
Are you generating your own lists too?

**Comment by markich** [comment]
> This is great @markich ! Thanks a lot for the work and effort. We are finishing a paper now and then we will talk to you to merge this. We need to test that our systems keep working in the same way with your modifications. Are you generating your own lists too?

Hey @eldraco!
Thanks for your feedback and kind words! :raised_hands:
Firstly, I'm wrapping up refactoring AIP.py into a well-structured class with tests in place to achieve a high coverage. Nevertheless, I would highly appreciate to get a sample input CSV file (the ones used by the **data_sorter** package) to cover at least border cases and verify that blocklists are being created accordingly, and with proper results.
Secondly, after submitting my latest local changes you all are going to be able to have a clearer sense of how the tool has been updated. Moreover, if the current state of the PR is confusing I can change it into a **DRAFT PR**.

Finally, I will also update the **PR** description because most of the action items detailed in the **UP NEXT** sections have been addressed so far.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+with open("version.py") as fp:
+    exec(fp.read(), version_globals)
+version = version_globals['__version__']
+
+setup(
+    name='AIP',
+    packages=find_packages(exclude=("tests",)),
+    version=version,
+    author='Stratosphere IPS',
+    author_email='stratosphereips@agents.fel.cvut.cz',
```
Comment (verovaleros) [inline setup.py:23]: stratosphere@aic.fel.cvut.cz

**Comment by jwackito** [comment]
Hi @markich! Thanks a lot for this work. Is more or less what I wanted to do but as my profile is not very developing oriented, I didn't know how to do it! Really this is great. I'm asking for permissions to accept the PR so we can move forward from there. I will contact by email to give you access to some example input files. Thanks again!

File: setup.py (hunk 1 above)
Comment (markich) [reply setup.py:23]: > [stratosphere@aic.fel.cvut.cz](mailto:stratosphere@aic.fel.cvut.cz)

@verovaleros @jwackito 
I'll draft a follow-up PR to update this, and enable the blocklist creation with mocked CSV files.

**Pull request merged**